api/apps/geoprocessing/src/modules/surface-cost/application/surface-cost-processor.ts

```diff
@@ -6,25 +6,30 @@ import { CostSurfaceJobInput } from '../cost-surface-job-input';
 
 import { CostSurfacePersistencePort } from '../ports/persistence/cost-surface-persistence.port';
 import { PuExtractorPort } from '../ports/pu-extractor/pu-extractor.port';
-import { ArePuidsAllowedPort } from '../ports/pu-validator/are-puuids-allowed.port';
+import { GetAvailablePlanningUnits } from '../ports/available-planning-units/get-available-planning-units';
 import { ShapefileConverterPort } from '../ports/shapefile-converter/shapefile-converter.port';
+import { lockPlanningUnits } from '@marxan/scenarios-planning-unit';
 
 @Injectable()
 export class SurfaceCostProcessor
   implements WorkerProcessor<CostSurfaceJobInput, true> {
   constructor(
     private readonly repo: CostSurfacePersistencePort,
     private readonly puExtractor: PuExtractorPort,
-    private readonly puValidator: ArePuidsAllowedPort,
+    private readonly availablePlanningUnits: GetAvailablePlanningUnits,
     private readonly shapefileConverter: ShapefileConverterPort,
   ) {}
 
   async process(job: Job<CostSurfaceJobInput, true>): Promise<true> {
     const geoJson = await this.shapefileConverter.convert(job.data.shapefile);
     const surfaceCosts = this.puExtractor.extract(geoJson);
-    const { errors } = await this.puValidator.validate(
-      job.data.scenarioId,
+    const planningUnitsIds = (
+      await this.availablePlanningUnits.get(job.data.scenarioId)
+    ).ids;
```

Comment:
linking to my earlier comment about breaking things too finely - ok, it's easier to test this, but after all in much of this we're after a list of scenario-planning-units, calculating a difference, and persisting the associated costs if there are no errors - aren't we introducing too many moving pieces here for something that we don't expect to need to change?

I would suggest to also call this `const scenarioPlanningUnitIds` to make it clear which set of PUs this const is expected to hold.

Reply:
https://github.com/Vizzuality/marxan-cloud/pull/225/commits/59f75722ee095cd6518244024ecadf51e85f618f